CoreInteractiveGraph/test/unit/src/au/gov/asd/tac/constellation/graph/interaction/gui/VisualGraphTopComponentNGTest.java

```diff
@@ -67,62 +81,256 @@ public static void tearDownClass() throws Exception {
                 throw e;
             }
         }
+
+        System.clearProperty("java.awt.headless");
+    }
+
+    @Test
+    public void testConstructor() {
+        System.out.println("testConstructor");
+        Platform.runLater(() -> {
+            // Mock variables
+            final GraphDataObject mockGDO = mock(GraphDataObject.class);
+            final FileObject mockFileObject = mock(FileObject.class);
+
+            when(mockGDO.isInMemory()).thenReturn(false);
+            when(mockGDO.isValid()).thenReturn(true);
+            when(mockGDO.getPrimaryFile()).thenReturn(mockFileObject);
+            try {
+                when(mockGDO.createFromTemplate(any(), anyString())).thenReturn(mockGDO);
+            } catch (IOException e) {
+                LOGGER.log(Level.WARNING, "Caught exception in mockGDO", e);
+            }
+
+            when(mockFileObject.getPath()).thenReturn("");
+
+            // Assert constructed correctely
+            VisualGraphTopComponent instance = new VisualGraphTopComponent();
+            instance.getGraphNode().setDataObject(mockGDO);
+            assertEquals(instance.getGraphNode().getDataObject(), mockGDO);
+            assertNotNull(instance);
+            assertNotNull(instance.getUndoRedo());
```

Comment:
if possible, it would be good to get the asserts out of the `Platform.runLater`. It's worth noting that these tests only keep an eye on asserts happening in the main thread and so if any of these asserts fail, they will not fail the test (it will come up in the logs that the asserts failed but the tests themselves will still suggest they have passed)

Generally speaking then, I would try to keep the code run in other threads do a bare minimum if possible, for the reasons above